GLAA.Web/Controllers/DefaultController.cs

```diff
@@ -83,27 +131,103 @@ protected virtual string GetLastViewPath(FormSection section)
         {
-            while (!FormDefinition.CanViewNextModel(section, id, model))
+            while (!FormDefinition.CanViewPage(section, id, model))
                 id++;
 
             return RedirectToAction(section, id);
         }
 
+        /// <summary>
+        /// Returns a redirect action for the first view following the supplied view that can be viewed 
+        /// based on the state of the parent view model.
+        /// </summary>
+        /// <typeparam name="T">The type of the parent view model.</typeparam>
+        /// <param name="section">The forms section.</param>
+        /// <param name="actionName">The name of the action to look for views after.</param>
+        /// <param name="model">The parent view model.</param>
+        /// <returns>A redirect <see cref="RedirectToActionResult"/> for the next possible view.</returns>
+        protected IActionResult RedirectToNextPossibleView<T>(FormSection section, string actionName, T model)
+            where T : IValidatable
+        {
+            var nextPage = FormDefinition.GetNextPossiblePage(section, actionName, model);
+
+            return RedirectToAction(section, nextPage.ActionName);
+        }
+
+        [System.Obsolete]
         protected virtual ActionResult RedirectToAction(FormSection section, int id)
         {
             return RedirectToAction("Part", section.ToString(), new {id});
         }
 
+        /// <summary>
+        /// Returns a <see cref="RedirectToActionResult"/> for the specified action name.
+        /// </summary>
+        /// <param name="section">The form section.</param>
+        /// <param name="actionName">The action name.</param>
+        /// <returns>A <see cref="RedirectToActionResult"/> for the specified action name.</returns>
+        protected virtual IActionResult RedirectToAction(FormSection section, string actionName)
+        {
+            return RedirectToAction(actionName, section.ToString());
+        }
+
+        [System.Obsolete]
         protected virtual ActionResult RedirectBackToAction(FormSection section, int id)
         {
             return RedirectToAction("Part", section.ToString(), new { id, back = true });
         }
 
+        /// <summary>
+        /// Returns a <see cref="RedirectToActionResult"/> for the specified action name with the 
+        /// <code>back</code> direction indicator in the query string.
+        /// </summary>
+        /// <param name="section">The form section.</param>
+        /// <param name="actionName">The action name.</param>
+        /// <returns>A <see cref="RedirectToActionResult"/> for the specified action name with the <code>back</code> indicator.</returns>
+        protected virtual IActionResult RedirectBackToAction(FormSection section, string actionName)
+        {
+            return RedirectToAction(actionName, section.ToString(), new {back = true});
```

Comment:
Half of me wonders whether we can just use javascript for the back button and do-away with a lot of this!

Reply:
That would make me SO HAPPY. The back logic is a massive pain, and the partial view that renders it makes my eyes hurt!